scripts/iscsictl.py

```diff
@@ -0,0 +1,392 @@
+        self.ssh.fdisk(sh.echo('-e', r'o\nn\np\n1\n\n\nw'), path)
+        self.ssh.losetup(loop, path)
+
+        is_in = self.find_loop(loop)
+        if not is_in:
+            raise Exception('fail to create loop device: %s / %s' %
+                            is_in)
+
+    def deploy(self):
+        """Deploy, configure and launch iSCSI target."""
+        self.zypper('iscsitarget')
+
+        self.create_loop(self.loop, self.path, self.size)
+
+        # Default configuration uses incoming autentication
+        lines = (
+            'IncomingUser user passwd',
+            'Target iqn.2015-01.qa.cloud.suse.de:%s' % self.iqn_id,
+            '"\tLun 0 Path=%s"' % self.loop,
+        )
+        self.append_cfg('/etc/ietd.conf', lines)
+
+        # Persist and start the service
+        self.service('iscsitarget', True)
+        self.ssh.rciscsitarget('restart')
+
+        # Remove the restart of iSCSI target service, so we can add it
+        # again at the end.
+        lines = ('rciscsitarget restart',)
+        self.remove_cfg('/etc/rc.d/boot.local', lines)
+
+        # Add in /etc/rc.d/boot.local
+        lines = (
+            'losetup %s %s' % (self.loop, self.path),
+            'rciscsitarget restart',
+        )
+        self.append_cfg('/etc/rc.d/boot.local', lines)
+
+        # Check if the device is exported
+        result = str(self.ssh.cat('/proc/net/iet/volume'))
+        if 'iqn.2015-01.qa.cloud.suse.de:%s' % self.iqn_id not in result:
+            raise Exception('Unable to deploy the iSCSI target')
+
+
+class Initiator(ISCSI):
+    """Define and manage an iSCSI initiator node."""
+
+    # For now, we are going to use the discovery option for iSCSI to
+    # populate the database.  This simplify the deployment of a basic
+    # iSCSI scenario, with a single target point and multiple
+    # initiators.
+
+    def __init__(self, ssh, target_ip):
+        """Initialize the Initiator instance with an ip and a mount point."""
+        super(Initiator, self).__init__(ssh)
+        self.target_ip = target_ip
+        # TODO XXX - This can be parametrized for other uses cases
+        self.name = 'iqn.2015-01.qa.cloud.suse.de:id01'
+
+    def deploy(self):
+        """Deploy, configure and persist an iSCSI initiator."""
+        self.zypper('open-iscsi')
+
+        # Default configuration only takes care of autentication
+        lines = (
+            'node.startup = automatic',
+            'node.session.auth.authmethod = CHAP',
+            'node.session.auth.username = user',
+            'node.session.auth.password = passwd',
+            'discovery.sendtargets.auth.authmethod = CHAP',
+            'discovery.sendtargets.auth.username = user',
+            'discovery.sendtargets.auth.password = passwd',
+        )
+        self.append_cfg('/etc/iscsid.conf', lines)
+
+        # Persist and start the service
+        self.service('open-iscsi', True)
+        self.ssh.sh('rcopen-iscsi', 'restart')
```

Comment:
Hm, maybe it was some my problem... I rebooted one node and than the iscsi device could not be found...